Neos.ContentRepository/Classes/Domain/ContentStream/ContentStreamIdentifier.php

```diff
@@ -12,36 +10,44 @@
  * source code.
  */
 
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Domain\ContentStream;
+
 use Neos\Cache\CacheAwareInterface;
 use Neos\Flow\Utility\Algorithms;
 use Neos\Flow\Annotations as Flow;
 
 /**
  * The ContentStreamIdentifier is the identifier for a Content Stream, which is
  * a central concept in the Event-Sourced CR introduced with Neos 5.0.
- *
- * @Flow\Proxy(false)
  */
-final class ContentStreamIdentifier implements \JsonSerializable, CacheAwareInterface
+#[Flow\Proxy(false)]
+final class ContentStreamIdentifier implements \JsonSerializable, CacheAwareInterface, \Stringable
 {
     /**
-     * @var string
+     * @var array<string,self>
      */
-    private $value;
+    private static array $instances = [];
+
+    private function __construct(
+        private string $value
+    ) {
+    }
 
-    private function __construct(string $value)
+    public static function instance(string $value): self
```

Comment:
I slightly prefer "constant" over "instance" (or something that communicates that this doesn't necessarily create a new instance). But probably that's just a matter of getting used to.

However, this method should be private IMO:
```suggestion
    private static function instance(string $value): self
```

Reply:
Already thought the same regarding instance() being private, especially as it is currently identical to fromString and could lead to confusion which one to use. I'd go with private, too.